### PR TITLE
feat: HA of ceresmeta

### PR DESCRIPTION
**Pull request by ShiKaiWi**
# Which issue does this PR close?

Closes #

# Rationale for this change
 Provide high availability for ceresmeta cluster.
 

# What changes are included in this PR?
- Elect the etcd leader as the ceresmeta leader.
- Watch the leader changes and elect a new one if the leader is not found.

# Are there any user-facing changes?

# How does this change test
Unit test + Test manually.

# Dependency
Requires the proto pr merged: https://github.com/CeresDB/ceresdbproto/pull/2

### CODE REVIEW

File: server/config/config.go [hunk 1]
```diff
@@ -164,10 +166,10 @@ func MakeConfigParser() (*Parser, error) {
 	fs.StringVar(&cfg.Log.File, "log-file", log.DefaultLogFile, "file for log output")
 
 	fs.Int64Var(&cfg.EtcdStartTimeoutMs, "etcd-start-timeout-ms", defaultEtcdStartTimeoutMs, "timeout for starting etcd server")
-	fs.Int64Var(&cfg.EtcdDialTimeoutMs, "etcd-dial-timeout-ms", defaultDailTimeoutMs, "timeout for dialing etcd server")
+	fs.Int64Var(&cfg.EtcdCallTimeoutMs, "etcd-dial-timeout-ms", defaultDailTimeoutMs, "timeout for dialing etcd server")
```
Comment (chunshao90) [inline server/config/config.go:169]: duplicate

File: server/etcdutil/config.go [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+	"fmt"
+	"net/url"
+	"os"
+
+	"github.com/tikv/pd/pkg/tempurl"
+	"go.etcd.io/etcd/server/v3/embed"
+)
+
+// NewTestSingleConfig is used to create an etcd config for the unit test purpose.
+func NewTestSingleConfig() *embed.Config {
```
Comment (chunshao90) [inline server/etcdutil/config.go:15]: move to server/etcdutil/config_test.go?

File: server/member/watch_leader.go [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+	"time"
+
+	"github.com/CeresDB/ceresmeta/pkg/log"
+	"github.com/CeresDB/ceresmeta/server/etcdutil"
+	clientv3 "go.etcd.io/etcd/client/v3"
+	"go.uber.org/zap"
+)
+
+const (
+	WatchLeaderFailInterval = time.Duration(200) * time.Millisecond
```
Comment (chunshao90) [inline server/member/watch_leader.go:15]: watchLeaderFailInterval

File: server/member/watch_leader.go [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+
+	waitReasonFailEtcd    = "fail to access etcd"
```
Comment (chunshao90) [inline server/member/watch_leader.go:17]: use const iota

File: server/member/watch_leader_test.go [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+
+func (ctx *mockWatchCtx) NewLease() clientv3.Lease {
+	return clientv3.NewLease(ctx.client)
+}
+
+func (ctx *mockWatchCtx) NewWatcher() clientv3.Watcher {
+	return clientv3.NewWatcher(ctx.client)
+}
+
+func TestWatchLeaderSingle(t *testing.T) {
```
Comment (chunshao90) [inline server/member/watch_leader_test.go:31]: split into some funcs?

File: server/etcdutil/config.go (hunk 2 above)
Comment (ShiKaiWi) [reply server/etcdutil/config.go:15]: The function can't be export if so.

File: server/member/watch_leader.go (hunk 4 above)
Comment (ShiKaiWi) [reply server/member/watch_leader.go:17]: The reason will be used to log but iota is not readable.

**Review by chunshao90 — APPROVED** [review]
LGTM

**Pull request merged**